src/openrct2-ui/UiContext.Linux.cpp

```diff
@@ -376,6 +376,29 @@ namespace OpenRCT2::Ui
             uiContext->ShowMessageBox(dialogMissingWarning);
             throw std::runtime_error(dialogMissingWarning);
         }
+
+        static std::string EscapePathForShell(const std::string& path)
+        {
+            char output[256] = { 0 };
+            char* outputPtr = output;
+            *outputPtr = '"';
+            outputPtr++;
+
+            for (uint32_t i = 0; i < path.length(); i++)
+            {
+                if (path[i] == '"')
+                {
+                    *outputPtr = '\\';
+                    outputPtr++;
+                }
+
+                *outputPtr = path[i];
+                outputPtr++;
+            }
+            *outputPtr = '"';
+
+            return std::string(output);
+        }
```

Comment:
If path is longer than 256 then you are going to exceed the fixed output buffer. I suggest creating a `std::string` of capacity 256 and appending to that.

Reply:
Since you're passing a std::string by reference anyway, there's no need to do iterate the buffer the C way. Instead, you can do something like:

```
size_t index = 0;
while ((index = path.find('"', index) != std::string::npos)
{
     path.replace(index, index + 1, 2, "\\\"");
     index += 2;
}
```